include/pqxx/transaction_base.hxx

```diff
@@ -389,6 +400,42 @@ public:
     return r[0].as<TYPE>();
   }
 
+  /// Perform query returning exactly one row, and convert its fields.
+  /** This is a convenient way of querying one row's worth of data, and
+   * converting its fields to a tuple of the C++-side types you specify.
+   *
+   * @throw unexpected_rows If the query did not return exactly 1 row.
+   * @throw usage_error If the number of columns in the result does not match
+   * the number of fields in the tuple.
+   */
+  template<typename... TYPE> std::tuple<TYPE...> query1(zview query)
```

Comment:
I think you should have `inline` on these

Reply:
And presumably `[[nodiscard]]`

Reply:
They're defined inline in the class, so the `inline` is implicit.  But you're right, they ought to be `[[nodiscard]]`.

Reply:
Ah, sorry. I missed that they were class members :-)